### PR TITLE
[LBSE] Treat root svg with clip path as not obscuring background

### PR DESCRIPTION
**Pull request by rwlbuis**
#### 2333404a4a2e271f6ea0f797f792442db05990de
<pre>
[LBSE] Treat root svg with clip path as not obscuring background
<a href="https://bugs.webkit.org/show_bug.cgi?id=265589">https://bugs.webkit.org/show_bug.cgi?id=265589</a>

Reviewed by Nikolas Zimmermann.

Treat root svg with clip path as not obscuring background. This is
needed since SVG documents and XML documents with SVG root nodes are
transparent, so in practise the non-clipped part would show up with
black background.

Tested by clip-path-on-svg-00*.svg.

* LayoutTests/platform/mac-sonoma-wk2-lbse-text/TestExpectations:
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::paintBoxDecorations):

Canonical link: <a href="https://commits.webkit.org/272486@main">https://commits.webkit.org/272486@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/c80e6481c8e4f0ea20ad6a355a4d7cdc3c1992f8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30983 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9657 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32672 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33499 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27993 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/31746 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12004 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6920 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27848 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31314 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8136 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27719 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6986 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7159 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/27598 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34838 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28211 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28068 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33297 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7192 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5266 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/31441 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8896 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7452 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7900 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7740 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/rendering/RenderView.cpp [hunk 1]
```diff
@@ -455,7 +455,7 @@ void RenderView::paintBoxDecorations(PaintInfo& paintInfo, const LayoutPoint&)
     if (RenderElement* rootRenderer = documentElement ? documentElement->renderer() : nullptr) {
         // The document element's renderer is currently forced to be a block, but may not always be.
         auto* rootBox = dynamicDowncast<RenderBox>(*rootRenderer);
-        rootFillsViewport = rootBox && !rootBox->x() && !rootBox->y() && rootBox->width() >= width() && rootBox->height() >= height();
+        rootFillsViewport = rootBox && (!rootBox->x() && !rootBox->y() && rootBox->width() >= width() && rootBox->height() >= height()) && !rootBox->hasClipPath();
```
Comment (nikolaszimmermann) [inline Source/WebCore/rendering/RenderView.cpp]: What are the implications for non-SVG renderers? Shouldn't CSS have the same issue for clip paths?

Comment (rwlbuis) [reply Source/WebCore/rendering/RenderView.cpp]: It looks like SVG is special, see this comment in RenderView.cpp:
// SVG documents and XML documents with SVG root nodes are transparent.

Comment (nikolaszimmermann) [reply Source/WebCore/rendering/RenderView.cpp]: Fair enough, but you are not checking specifically for SVG?

Comment (rwlbuis) [reply Source/WebCore/rendering/RenderView.cpp]: I am in the new patch :)

**Review by nikolaszimmermann — APPROVED** [review]
Thanks r=me!

**Pull request merged**